packages/bp/src/orchestrator/messaging-server.ts

```diff
@@ -91,3 +85,47 @@ export const startMessagingServer = async (opts: Partial<MessagingServerOptions>
     })
   })
 }
+
+export const runMessagingMigration = async (
+  cmd: 'up' | 'down',
+  target: string,
+  dry: boolean | undefined
+): Promise<sdk.MigrationResult> => {
+  const env = {
+    NODE_ENV: process.env.NODE_ENV,
+    NODE_OPTIONS: '',
+    DATABASE_URL: process.core_env.DATABASE_URL || `${process.PROJECT_LOCATION}/data/storage/core.sqlite`,
+    DATABASE_POOL: process.env.DATABASE_POOL,
+    SKIP_LOAD_ENV: 'true',
+    SPINNED: 'true',
+
+    MIGRATE_CMD: cmd,
+    MIGRATE_TARGET: target,
+    MIGRATE_DRYRUN: dry ? 'true' : 'false'
+  }
+  const proc = await spawnMessagingProcess(env)
+
+  return new Promise(resolve => {
+    proc.on('exit', code => {
+      if (code === 0) {
+        resolve({ success: true, message: `Messaging database updated successfully to version ${target}` })
+      } else {
+        resolve({ success: false, message: `Failed to update messaging database to version ${target}` })
```

Comment:
I don't think we should resolve when this happens. When Botpress fails to migrate using --auto-migrate, the process simply exits

Reply:
We resolve but we send success: false. Otherwise if we reject the promise we don't get a clear error message